### PR TITLE
[FLINK-36336] Remove `flink-java` module

### PR DESCRIPTION
**Pull request by codenohup**
## What is the purpose of the change

The dataset APIs have been deprecated, and we need to remove the `flink-java` module.

This goal of this PR is remove `flink-java` module, deprecated ExecutionMode, dataset and scala example in documents.

## Brief change log

- remove `flink-java` module
- remove deprecated ExecutionMode in flink-core module
- remove dataset and scala example in documents

## Verifying this change

This change is a code cleanup without any test coverage.
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by codenohup** [comment]
@flinkbot run azure

File: flink-datastream-api/src/main/java/org/apache/flink/datastream/api/ExecutionEnvironment.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ static ExecutionEnvironment getInstance() throws ReflectiveOperationException {
     RuntimeExecutionMode getExecutionMode();
 
     /** Set the execution mode for this environment. */
-    ExecutionEnvironment setExecutionMode(RuntimeExecutionMode runtimeMode);
+    void setExecutionMode(RuntimeExecutionMode runtimeMode);
```
Comment (reswqa) [inline flink-datastream-api/src/main/java/org/apache/flink/datastream/api/ExecutionEnvironment.java:52]: Why change this return type?

File: flink-datastream/src/main/java/org/apache/flink/datastream/impl/ExecutionEnvironmentImpl.java [hunk 2]
```diff
@@ -147,10 +147,9 @@ public RuntimeExecutionMode getExecutionMode() {
     }
 
     @Override
-    public ExecutionEnvironment setExecutionMode(RuntimeExecutionMode runtimeMode) {
+    public void setExecutionMode(RuntimeExecutionMode runtimeMode) {
         checkNotNull(runtimeMode);
         configuration.set(ExecutionOptions.RUNTIME_MODE, runtimeMode);
-        return this;
```
Comment (reswqa) [inline flink-datastream/src/main/java/org/apache/flink/datastream/impl/ExecutionEnvironmentImpl.java:153]: ?

**Review by reswqa — COMMENTED** [review]
Thanks for this PR, I have only minor comments, PTAL.

**Comment by codenohup** [comment]
> Thanks for this PR, I have only minor comments, PTAL.

Hi, @reswqa 
Thank you for your carefully review, i have made adjustments according to your suggestions, and have rebased to master branch.

**Comment by codenohup** [comment]
@flinkbot run azure

**Comment by codenohup** [comment]
@flinkbot run azure

**Pull request merged**